CvGameCoreDLL_Expansion2/Lua/CvLuaPlayer.cpp

```diff
@@ -11831,36 +11831,29 @@ int CvLuaPlayer::lGetRecommendedWorkerPlots(lua_State* L)
 	if(pWorkerUnit == NULL)
 		return 0;
 
-	//fake the reachable plots, ignore all other workers
-	map<int, ReachablePlots> allplots;
-	SPathFinderUserData data(pWorkerUnit, 0, 3);
-	allplots[pWorkerUnit->GetID()] = GC.GetPathFinder().GetPlotsInReach(pWorkerUnit->plot(), data);
+	BuilderDirective assignedDirective = pkPlayer->GetBuilderTaskingAI()->GetAssignedDirective(pWorkerUnit);
 
-	BuilderDirective aDirective = pkPlayer->GetBuilderTaskingAI()->EvaluateBuilder(pWorkerUnit,allplots);
-	if(aDirective.m_eDirective == BuilderDirective::NUM_DIRECTIVES)
-		return 0;
-
-	//don't look at me ... this is just for stupid lua
-	const size_t cuiMaxDirectives = 3;
-	const size_t cuiDirectiveSize = 1;
+	if (assignedDirective.m_eBuild != NO_BUILD)
+	{
+		lua_createtable(L, 0, 0);
+		int iCount = 1;
 
-	lua_createtable(L, cuiMaxDirectives, 0);
-	int iPositionIndex = lua_gettop(L);
-	int i = 1;
+		CvPlot* pPlot = GC.getMap().plot(assignedDirective.m_sX, assignedDirective.m_sY);
+		if (pPlot)
+		{
+			lua_createtable(L, 0, 0);
+			const int t = lua_gettop(L);
+			CvLuaPlot::Push(L, GC.getMap().plot(assignedDirective.m_sX, assignedDirective.m_sY));
+			lua_setfield(L, t, "plot");
+			lua_pushinteger(L, assignedDirective.m_eBuild);
+			lua_setfield(L, t, "buildType");
+			lua_rawseti(L, -2, iCount++);
```

Comment:
iCount is always 2

Reply:
I think this was copied from somewhere, lot's of dumb code surrounding the Lua code integration.